ui-v2/src/components/ui/run-state-change-dialog/run-state-change-dialog.tsx

```diff
@@ -0,0 +1,205 @@
+import type { components } from "@/api/prefect";
+import { Button } from "@/components/ui/button";
+import {
+	Dialog,
+	DialogContent,
+	DialogHeader,
+	DialogTitle,
+} from "@/components/ui/dialog";
+import {
+	Form,
+	FormControl,
+	FormField,
+	FormItem,
+	FormLabel,
+	FormMessage,
+} from "@/components/ui/form";
+import {
+	Select,
+	SelectContent,
+	SelectGroup,
+	SelectItem,
+	SelectTrigger,
+	SelectValue,
+} from "@/components/ui/select";
+import { StateBadge } from "@/components/ui/state-badge";
+import { Textarea } from "@/components/ui/textarea";
+import { RUN_STATES, type RunStates } from "@/utils/states";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useState } from "react";
+import { useForm } from "react-hook-form";
+import { z } from "zod";
+
+const formSchema = z.object({
+	state: z.enum(
+		Object.keys(RUN_STATES) as [
+			components["schemas"]["StateType"],
+			...components["schemas"]["StateType"][],
+		],
+	),
+	message: z.string().optional().default(""),
+});
+
+export type RunStateFormValues = z.infer<typeof formSchema>;
+
+type RunState = {
+	id: string;
+	type: RunStates;
+	name: string;
+	message?: string | null;
+};
+
+export type RunStateDialogProps = {
+	currentState: RunState;
+	open: boolean;
+	onOpenChange: (open: boolean) => void;
+	title: string;
+	onSubmitChange: (values: RunStateFormValues) => Promise<void>;
+};
+
+export const RunStateChangeDialog = ({
+	currentState,
+	open,
+	onOpenChange,
+	title,
+	onSubmitChange,
+}: RunStateDialogProps) => {
+	const [isSubmitting, setIsSubmitting] = useState(false);
+
+	const form = useForm({
+		resolver: zodResolver(formSchema),
+		defaultValues: {
+			state:
+				(Object.keys(RUN_STATES).find(
+					(state) => state !== currentState.type,
+				) as RunStates) || "PENDING",
```

Comment:
Can this start in an unselected state (i.e. start with no value)?

Reply:
updated in [bea96e7](https://github.com/PrefectHQ/prefect/pull/17704/commits/bea96e7ffe9923521e53cdd00977134bf69d8512)
<img width="509" alt="image" src="https://github.com/user-attachments/assets/99d10d52-3e85-4f15-86a0-5e78091dd428" />